src/redisraft.c

```diff
@@ -1008,6 +1008,104 @@ static int cmdRaftRandom(RedisModuleCtx *ctx,
     return REDISMODULE_OK;
 }
 
+#ifdef HAVE_TLS
+/* Callback for passing a keyfile password stored as a char * to OpenSSL,  copied from redis */
+static int tlsPasswordCallback(char *buf, int size, int rwflag, void *u) {
+    UNUSED(rwflag);
+
+    const char *pass = u;
+    size_t pass_len;
+
+    if (!pass) return -1;
+    pass_len = strlen(pass);
+    if (pass_len > (size_t) size) return -1;
+    memcpy(buf, pass, pass_len);
+
+    return (int) pass_len;
+}
+
+SSL_CTX *generateSSLContext(RedisModuleCtx *ctx, RedisRaftCtx *rr) {
+    SSL_CTX *ssl_ctx = SSL_CTX_new(SSLv23_client_method());
+    if (!ssl_ctx) {
+        RedisModule_Log(ctx, REDIS_WARNING, "REDIS_SSL_CTX_CREATE_FAILED");
+//        if (error) *error = REDIS_SSL_CTX_CREATE_FAILED;
+        return NULL;
+    }
+
+    SSL_CTX_set_options(ssl_ctx, SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3);
+    SSL_CTX_set_verify(ssl_ctx, SSL_VERIFY_PEER, NULL);
+
+    if ((rr->config->tls_cert != NULL && rr->config->tls_key == NULL) ||
+            (rr->config->tls_key != NULL && rr->config->tls_cert == NULL)) {
+        RedisModule_Log(ctx, REDIS_WARNING, "REDIS_SSL_CTX_CERT_KEY_REQUIRED");
+//        if (error) *error = REDIS_SSL_CTX_CERT_KEY_REQUIRED;
+        goto error;
+    }
+
+    if (!SSL_CTX_load_verify_locations(ssl_ctx, rr->config->tls_ca_cert, NULL)) {
```

Comment:
Don't we want to print TLS errors here and in similar checks down below?

Reply:
don't understand the Q? like an openssl error function to get an error string?  I don't see those that do it.  otherwise we have our own logging (now converted to `LOG_ERROR()`)

Reply:
When `SSL_CTX_load_verify_locations()` fails, we won't see what is the error. Why don't we print the error? Something like 
```
unsigned long e = ERR_peek_last_error();
LOG_ERROR("SSL_CTX_load_verify_locations(): %s", ERR_reason_error_string(e))
```

Reply:
ok, done.